domain/src/main/java/org/oppia/domain/util/StateRetriever.kt

```diff
@@ -287,26 +289,21 @@ class StateRetriever @Inject constructor(
       is Int -> return interactionObjectBuilder.setSignedInt(customizationArgValue).build()
       is Double -> return interactionObjectBuilder.setReal(customizationArgValue).build()
       is Boolean -> return interactionObjectBuilder.setBoolValue(customizationArgValue).build()
-      else -> {
-        val customizationArgValueTemp: ArrayList<*> =
-          Gson().fromJson(customizationArgValue.toString(), ArrayList::class.java)
-        if (customizationArgValueTemp is List<*> && customizationArgValueTemp.size > 0) {
-          return interactionObjectBuilder.setSetOfHtmlString(
-            createStringList(customizationArgValueTemp)
-          ).build()
-        }
+      is JSONArray -> if (customizationArgValue.length() > 0) {
+        return interactionObjectBuilder.setSetOfHtmlString(
+          createStringList(JsonParser().parseJsonArrayToMutableListOfString(customizationArgValue))
+        ).build()
       }
     }
     return InteractionObject.getDefaultInstance()
   }
 
   @Suppress("UNCHECKED_CAST") // Checked cast in the if statement
-  private fun createStringList(value: List<*>): StringList {
-    val stringList = mutableListOf<String>()
-    if (value[0] is String) {
-      stringList.addAll(value as List<String>)
+  private fun createStringList(stringList: MutableList<String>): StringList {
```

Comment:
I think this can be just a ``List``.

Reply:
Also, could we just take a JSONArray here, instead, to simplify?

Reply:
Done